src/main/java/com/arpnetworking/tsdcore/model/AggregationMessage.java

```diff
@@ -103,6 +109,87 @@ public ByteString serializeToByteString() {
         return sizePrefix.result().concat(bs);
     }
 
+
+    /**
+     * Deserialize message from {@link Buffer}.
+     *
+     * @param data The {@link Buffer} containing the serialized message.
+     * @return The deserialized {@link AggregationMessage} or absent if
+     * the {@link Buffer} could not be deserialized.
+     */
+    public static Optional<AggregationMessage> deserialize(final ByteString data) {
```

Comment:
Is there really no utility that would do this for you?! wild

Reply:
No... we write this function in CAGG so I basically copied that part from CAGG.